From ab37d8d8cf9f08ea203e70059b6db6d9985874ad Mon Sep 17 00:00:00 2001
From: alex <surkov.alex@gmail.com>
Date: Thu, 2 Mar 2017 11:37:04 -0500
Subject: [PATCH] remove SD resolution

---
 xbmc/utils/AMLUtils.cpp | 36 ------------------------------------
 1 file changed, 36 deletions(-)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index cd19ec7..e9ac90d 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -354,42 +354,6 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
     res->fRefreshRate = 60;
     res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
   }
-  else if ((StringUtils::EqualsNoCase(fromMode, "480cvbs")) || (StringUtils::EqualsNoCase(fromMode, "480i")))
-  {
-    res->iWidth = 720;
-    res->iHeight= 480;
-    res->iScreenWidth = 720;
-    res->iScreenHeight= 480;
-    res->fRefreshRate = 60;
-    res->dwFlags = D3DPRESENTFLAG_INTERLACED;
-  }
-  else if ((StringUtils::EqualsNoCase(fromMode, "576cvbs")) || (StringUtils::EqualsNoCase(fromMode, "576i")))
-  {
-    res->iWidth = 720;
-    res->iHeight= 576;
-    res->iScreenWidth = 720;
-    res->iScreenHeight= 576;
-    res->fRefreshRate = 50;
-    res->dwFlags = D3DPRESENTFLAG_INTERLACED;
-  }
-  else if (StringUtils::EqualsNoCase(fromMode, "480p"))
-  {
-    res->iWidth = 720;
-    res->iHeight= 480;
-    res->iScreenWidth = 720;
-    res->iScreenHeight= 480;
-    res->fRefreshRate = 60;
-    res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
-  }
-  else if (StringUtils::EqualsNoCase(fromMode, "576p"))
-  {
-    res->iWidth = 720;
-    res->iHeight= 576;
-    res->iScreenWidth = 720;
-    res->iScreenHeight= 576;
-    res->fRefreshRate = 50;
-    res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
-  }
   else if (StringUtils::EqualsNoCase(fromMode, "720p") || StringUtils::EqualsNoCase(fromMode, "720p60hz"))
   {
     res->iWidth = 1280;
-- 
1.9.1

